[api-docs] Various fixes

sean-gilliam opened the pull request:
This PR contains various fixes including filling in TBD placeholders,
fixed numerous copyright headers and converted a repetitive class to a
T4 template.

Danthar commented:
public api has changed. Dont think that was intentional :P

Aaronontheweb commented:
@Danthar @sean-gilliam marking a method with an `OSBSOLETE` attribute can trigger a public API change, since attribute metadata does get factored into that.

btw, what are the T4 templates for?

sean-gilliam commented:
It was just for one class. There was a lot of classes with the same boilerplate and signatures like Class<T, T1>, Class <T, T1, T2>,Class<T, T1, T2,  ...>. Just a lot of repetition. Changed it to make it easier to document.

The API change came in with one of that classes' methods. At the very end, it changes one of its parameters names. I fixed it to match the other method signatures. It's just a Utility class that user's prb wouldn't care about.

Aaronontheweb commented on src/core/Akka.API.Tests/CoreAPISpec.ApproveCore.approved.txt:
```diff
@@ -4639,7 +4639,7 @@ namespace Akka.Tools.MatchHandler
     {
         public PartialHandlerArgumentsCapture() { }
-        public bool Handle(T message) { }
+        public bool Handle(T value) { }
```
Api change is just a name of the parameter, which shouldn't matter really.
